File: graphedu/common/models/vo/systemv2/dept.py
"""部门管理相关 VO 模型 (View Objects - 响应模型)

职责：
1. 定义 API 响应的数据结构
2. 配置序列化规则（from_attributes=True 支持从 ORM 对象创建）
"""

from datetime import datetime

from pydantic import Field

from graphedu.common.models.vo.base import VO


class DeptTreeVO(VO):
    """部门树节点 VO"""

    dept_id: int = Field(description="部门ID")
    dept_name: str = Field(description="部门名称")
    parent_id: int = Field(description="父部门ID")
    dept_key: str = Field(description="部门编码")
    leader: str | None = Field(default=None, description="负责人")
    phone: str | None = Field(default=None, description="联系电话")
    email: str | None = Field(default=None, description="联系邮箱")
    status: str = Field(description="对照sys_data_status（0正常 1停用 2已删除）")
    sort_order: int = Field(description="显示顺序")
    create_time: datetime | None = Field(default=None, description="创建时间")
    has_children: bool = Field(default=False, description="是否有子部门")
    children: list["DeptTreeVO"] | None = Field(default=None, description="子部门列表，一般推荐自己组装")


class DeptDetailVO(VO):
    """部门详细信息 VO"""

    dept_id: int = Field(description="部门ID")
    parent_id: int = Field(description="父部门ID")
    dept_name: str = Field(description="部门名称")
    dept_key: str = Field(description="部门编码")
    leader: str | None = Field(default=None, description="负责人")
    phone: str | None = Field(default=None, description="联系电话")
    email: str | None = Field(default=None, description="联系邮箱")
    status: str = Field(description="对照sys_data_status（0正常 1停用 2已删除）")
    sort_order: int = Field(description="显示顺序")
    create_by: int | None = Field(default=None, description="创建者ID")
    create_time: datetime | None = Field(default=None, description="创建时间")
    update_by: int | None = Field(default=None, description="更新者ID")
    update_time: datetime | None = Field(default=None, description="更新时间")
    remark: str | None = Field(default=None, description="备注")


class DeptSimpleVO(VO):
    """部门简要信息 VO"""

    dept_id: int = Field(description="部门ID")
    dept_name: str = Field(description="部门名称")
    dept_key: str = Field(description="部门编码")
    parent_id: int = Field(description="父部门ID")


class DeptInfoVO(VO):
    """部门信息 VO (用于列表展示)"""

    dept_id: int = Field(description="部门ID")
    parent_id: int = Field(description="父部门ID")
    dept_name: str = Field(description="部门名称")
    dept_key: str = Field(description="部门编码")
    leader: str | None = Field(default=None, description="负责人")
    phone: str | None = Field(default=None, description="联系电话")
    email: str | None = Field(default=None, description="联系邮箱")
    status: str = Field(description="对照sys_data_status（0正常 1停用 2已删除）")
    sort_order: int = Field(description="显示顺序")
    create_time: datetime | None = Field(default=None, description="创建时间")
    remark: str | None = Field(default=None, description="备注")


# 更新前向引用
DeptTreeVO.model_rebuild()

File: graphedu-ui/src/types/api/system/dept.ts
/**
 * 部门管理相关类型定义
 * 对应后端：graphedu/common/models/dto/dept.py 和 vo/dept.py
 */

// ============================================================================
// 请求 DTO 类型
// ============================================================================

/**
 * 部门查询请求
 */
export interface DeptQueryDTO {
  /** 部门ID */
  deptId?: number
  /** 部门名称（模糊查询） */
  deptName?: string
  /** 父部门ID，0或None表示根节点 */
  parentId?: number
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status?: '0' | '1'
}

/**
 * 创建部门请求
 */
export interface DeptCreateDTO {
  /** 父部门ID（0表示根节点） */
  parentId?: number
  /** 部门名称 */
  deptName: string
  /** 部门编码（唯一标识） */
  deptKey: string
  /** 负责人 */
  leader?: string
  /** 联系电话 */
  phone?: string
  /** 联系邮箱 */
  email?: string
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status?: '0' | '1'
  /** 显示顺序 */
  sortOrder?: number
  /** 备注 */
  remark?: string
}

/**
 * 更新部门请求
 */
export interface DeptUpdateDTO {
  /** 部门ID */
  deptId: number
  /** 父部门ID */
  parentId?: number
  /** 部门名称 */
  deptName?: string
  /** 部门编码 */
  deptKey?: string
  /** 负责人 */
  leader?: string
  /** 联系电话 */
  phone?: string
  /** 联系邮箱 */
  email?: string
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status?: '0' | '1'
  /** 显示顺序 */
  sortOrder?: number
  /** 备注 */
  remark?: string
}

/**
 * 修改部门状态请求
 */
export interface DeptStatusChangeDTO {
  /** 部门ID */
  deptId: number
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status: '0' | '1'
}

// ============================================================================
// 响应 VO 类型
// ============================================================================

/**
 * 部门树节点
 */
export interface DeptTreeVO {
  /** 部门ID */
  deptId: number
  /** 部门名称 */
  deptName: string
  /** 父部门ID */
  parentId: number
  /** 部门编码 */
  deptKey: string
  /** 负责人 */
  leader?: string
  /** 联系电话 */
  phone?: string
  /** 联系邮箱 */
  email?: string
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status: string
  /** 显示顺序 */
  sortOrder: number
  /** 创建时间 */
  createTime?: string
  /** 是否有子部门 */
  hasChildren?: boolean
  /** 子部门列表 */
  children?: DeptTreeVO[]
}

/**
 * 部门详细信息
 */
export interface DeptDetailVO {
  /** 部门ID */
  deptId: number
  /** 父部门ID */
  parentId: number
  /** 部门名称 */
  deptName: string
  /** 部门编码 */
  deptKey: string
  /** 负责人 */
  leader?: string
  /** 联系电话 */
  phone?: string
  /** 联系邮箱 */
  email?: string
  /** 部门状态，对照sys_data_status（0正常 1停用） */
  status: string
  /** 显示顺序 */
  sortOrder: number
  /** 创建者ID */
  createBy?: number
  /** 创建时间 */
  createTime?: string
  /** 更新者ID */
  updateBy?: number
  /** 更新时间 */
  updateTime?: string
  /** 备注 */
  remark?: string
}

/**
 * 部门简要信息
 */
export interface DeptSimpleVO {
  /** 部门ID */
  deptId: number
  /** 部门名称 */
  deptName: string
  /** 部门编码 */
  deptKey: string
  /** 父部门ID */
  parentId: number
}
